pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test: TestLogic_inverted_filter_geospatial failed

pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test.TestLogic_inverted_filter_geospatial [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8344589?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8344589?buildTab=artifacts#/) on master @ [6116ab435ee2cc37af34c99e28dec51b6cfa0eb4](https://github.com/cockroachdb/cockroach/commits/6116ab435ee2cc37af34c99e28dec51b6cfa0eb4):


```
=== RUN   TestLogic_inverted_filter_geospatial
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8ee417166775d051de542acde5a083e2/logTestLogic_inverted_filter_geospatial2229431385
    test_log_scope.go:79: use -show-logs to present logs inline
    logic.go:3233: 
        
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9502/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_filter_geospatial:52: SELECT k FROM geo_table WHERE 'POINT(3.0 3.0)'::geometry && geom ORDER BY k
        expected success, but found
        (0A000) this box2d comparison operator is experimental
        settings.go:33: in checkExperimentalBox2DComparisonOperatorEnabled()
    logic.go:2907: 
         pq: this box2d comparison operator is experimental
[15:11:13] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9502/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_filter_geospatial with config fakedist-vec-off: 9 tests, 2 failures
    logic.go:4025: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9502/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_filter_geospatial:58: error while processing
    logic.go:4025: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9502/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_filter_geospatial:58: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8ee417166775d051de542acde5a083e2/logTestLogic_inverted_filter_geospatial2229431385
--- FAIL: TestLogic_inverted_filter_geospatial (1.37s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_inverted_filter_geospatial.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23480

--- a/pkg/sql/logictest/testdata/logic_test/inverted_filter_geospatial
+++ b/pkg/sql/logictest/testdata/logic_test/inverted_filter_geospatial
@@ -49,7 +49,7 @@ SELECT k FROM geo_table WHERE ST_DWithin('POINT(2.5 2.5)'::geometry, geom, 1) OR
 statement ok
 SET CLUSTER SETTING sql.spatial.experimental_box2d_comparison_operators.enabled = on
 
-query I
+query I retry
 SELECT k FROM geo_table WHERE 'POINT(3.0 3.0)'::geometry && geom ORDER BY k
 ----
 3

